components/modal/modals/ServerIdentity.tsx

```diff
@@ -0,0 +1,345 @@
+              compact="icon"
+              palette="plain"
+              disabled={!props.member.nickname}
+              onClick={() => {
+                props.member.edit({ remove: ["Nickname"] });
+                setNickname("");
+              }}
+            >
+              <BiRegularX size={24} />
+            </Button>
+          </Row>
+        </Column>
+
+        <Row>
+          <Column>
+            <Typography variant="label">
+              {t("app.special.popovers.server_identity.avatar")}
+            </Typography>
+
+            <Column align="center">
+              <AvatarEdit member={props.member}></AvatarEdit>
+
+              <div onclick={openFilePicker} style={{ cursor: "pointer" }}>
+                <Show
+                  when={props.member.avatarURL === props.member.user?.avatarURL}
+                >
+                  <CategoryButton onClick={() => {}}>
+                    {t("app.settings.actions.upload")}
+                    <input
+                      ref={fileInputRef!}
+                      onChange={handleFileSelect}
+                      type="file"
+                      accept="image/*"
+                      multiple={false}
+                      style={{ display: "none" }}
+                      max={4_000_000}
+                    />
+                  </CategoryButton>
+                </Show>
+
+                <Show
+                  when={props.member.avatarURL != props.member.user?.avatarURL}
+                >
+                  <CategoryButton
+                    onClick={() => {
+                      props.member.edit({ remove: ["Avatar"] });
+                    }}
+                  >
+                    {t("app.settings.actions.remove")}
+                  </CategoryButton>
+                </Show>
+              </div>
+
+              <Typography variant="small">
+                {t("app.settings.actions.max_filesize", {
+                  filesize: "4.00 MB",
+                })}
+              </Typography>
+            </Column>
+          </Column>
+
+          <Column grow style={{ "padding-left": "0.75em" }}>
+            <Typography variant="label">
+              {t("app.special.modals.actions.preview")}
+            </Typography>
+
+            <Preview member={props.member}></Preview>
+          </Column>
+        </Row>
+      </>
+    ),
+  };
+};
+
+export default ServerIdentity;
+
+function AvatarEdit(props: { member: ServerMember }) {
+  const [isHovered, setIsHovered] = createSignal(false);
+
+  function onMouseEnter() {
+    setIsHovered(true);
+  }
+
+  function onMouseLeave() {
+    setIsHovered(false);
+  }
+
+  const user = () => userInformation(props.member.user, props.member);
+
+  let fileInputRef: HTMLInputElement | null = null;
+  const openFilePicker = () => {
+    fileInputRef?.click();
+  };
+
+  const handleFileSelect = async (event: Event) => {
+    const file = event.target?.files[0];
+    const json = await uploadToAutumn(file);
```

Comment:
The `target` property in `Event` isn't guaranteed to be `HTMLInputElement`, so typescript doesn't know whether the `files` property actually exists or not and throws a error.

```suggestion
  const handleFileSelect = async (event: Event & {target: HTMLInputElement}) => {
```